robot/Cumulus/tests/browser/recurring_donations/zedit_effective_date_erd_record.robot

```diff
@@ -0,0 +1,106 @@
+     ${DATE}=                              Get current date      result_format=%Y-%m-%d %H:%M:%S.%f      increment=30 days
+     ${DATE_TO_UPDATE} =                   Convert Date          ${DATE}                         result_format=%Y-%m-%d
+     ${EFFECTIVE_MODIFIED_DATE}=           Get current date      result_format=%-d/%-m/%Y      increment=0 days
+     ${CURRDATE}=                          Get Current Date      result_format=datetime
+     ${CURRENTVALUE} =                     Evaluate              (${CURRDATE.month-1}) * 100
+     ${CURRENTVALUE_EDITED}=               Evaluate              (${CURRDATE.month}*100) + 150
+     Set Suite Variable  ${CURRENTVALUE}
+     Set Suite Variable  ${DATE}
+     Set Suite Variable  ${EFFECTIVE_MODIFIED_DATE}
+     Set Suite Variable  ${CURRENTVALUE_EDITED}
+     Set Suite Variable  ${EFFECTIVE_DATE_INITIAL}
+     Set Suite Variable  ${DATE_TO_UPDATE}
+
+     &{contact1_fields}=              Create Dictionary          Email=rd2tester@example.com
+     &{recurringdonation_fields} =	 Create Dictionary          Name=ERDTest1
+     ...                                                         npe03__Installment_Period__c=Monthly
+     ...                                                         npe03__Amount__c=100
+     ...                                                         npe03__Open_Ended_Status__c=${TYPE}
+     ...                                                         Status__c=Active
+     ...                                                         Day_of_Month__c=${DAY_OF_MONTH}
+     ...                                                         InstallmentFrequency__c=${FREQUENCY}
+     ...                                                         PaymentMethod__c=${METHOD}
+
+     Setupdata   contact             ${contact1_fields}          recurringdonation_data=${recurringdonation_fields}
+
+
+*** Variables ***
+${FREQUENCY}  1
+${DAY_OF_MONTH}  15
+${AMOUNT_TO_UPDATE}  150
+${METHOD}  Credit Card
+${TYPE}    Open
+
+*** Test Cases ***
+
+Edit An Enhanced Recurring donation record of type open
+    [Documentation]               After creating an open recurring donation using API, The test ensures that when the record
+     ...                          is edited and effective date is updated to next month , Verifies that future schedule is shown
+     ...                          with the right values reflected. Verifies the current year and future year schedules are updated
+     ...                          Verifies the opportunity status
+    [tags]                                   unstable               W-041167            feature:RD2
+
+    Go To Page                               Details
+    ...                                      npe03__Recurring_Donation__c
+    ...                                      object_id=${data}[contact_rd][Id]
+    ${rd_id}                                 Save Current Record ID For Deletion       npe03__Recurring_Donation__c
```

Comment:
when u r already creating the record with api, u dont need to store session record again

Reply:
deleted it. I modified it to  use the recurring donation ID direction from the data object to query form the database